src/bake/client/data-view.ts

```diff
@@ -1,10 +1,10 @@
 import { td, te } from "../shared";
 
 export class DataViewReader {
-  view: DataView<ArrayBuffer>;
```

Comment:
Why was this changed?

Reply:
It's a TS error when using `Buffer.prototype.buffer` because `DataView<ArrayBuffer>`  means it's not a `SharedArrayBuffer`

Reply:
yes. why is a sharedarraybuffer used here? 

additionally, an `as` assertion is placed elsewhere because `SharedArrayBuffer` isn't supported by the TextDecoder API. so this type doesn't actually work with shared array buffers.

Reply:
it's not used here. it just hypotehtically could be because Buffer.prototype.buffer can return a SharedArrayBuffer.

TextDecoder.prototype.decode does support SharedArrayBuffer afaik.

Reply:
In another comment you wrote:

> `TextDecoder.prototype.decode`'s type asserts that it is `ArrayBuffer` and not `SharedArrayBuffer`

Reply:
yeah, the type is wrong:

https://github.com/oven-sh/WebKit/blob/1517d271d44e1764a5df650052baa4d81eb4ca81/Source/WebCore/dom/TextDecoder.idl#L43